File: CitasAPI/src/main/java/citas_service_nuevo/model/Usuario.java
package citas_service_nuevo.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "usuario")
public class Usuario {

    @Id
    // Sin @GeneratedValue para usar el ID real que viene del Login
    @Column(name = "id_usuario")
    private Long id;

    @Column
    private String nombre;

    @Column
    private String apellido;

    @Column(name = "fecha_nacimiento")
    private LocalDateTime fechaNacimiento;

    @Column
    private String correo;

    @Column
    private String telefono;

    @Column
    private String contrasena;

    // [CORRECCIÓN] Eliminamos 'private Rol rol' porque CitasAPI no lo necesita.

    public Usuario() {
    }

    // --- Getters y Setters ---

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public LocalDateTime getFechaNacimiento() {
        return fechaNacimiento;
    }

    public void setFechaNacimiento(LocalDateTime fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }

    public String getCorreo() {
        return correo;
    }

    public void setCorreo(String correo) {
        this.correo = correo;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public String getContrasena() {
        return contrasena;
    }

    public void setContrasena(String contrasena) {
        this.contrasena = contrasena;
    }
}

File: historial-service/bin/main/com/clinica/api/historial_service/model/Usuario.kt
package com.clinica.api.historial_service.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.LocalDate

@Entity
@Table(name = "usuario")
data class Usuario(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_usuario")
    var id: Long? = null,

    var nombre: String,
    var apellido: String,

    @Column(name = "fecha_nacimiento")
    var fechaNacimiento: LocalDate,

    var correo: String,
    var telefono: String?,
    var contrasena: String,

    @Column(name = "id_rol")
    var rolId: Long
)
